dot/core/messages_test.go

```diff
@@ -112,295 +58,362 @@ func TestService_TransactionsCount(t *testing.T) {
 	}
 }
 
-func TestServiceHandleTransactionMessage(t *testing.T) {
-	testEmptyHeader := types.NewEmptyHeader()
-	testExtrinsic := []types.Extrinsic{{1, 2, 3}}
+func Test_Service_HandleTransactionMessage(t *testing.T) {
+	t.Parallel()
 
-	ctrl := gomock.NewController(t)
-	runtimeMock := NewMockRuntimeInstance(ctrl)
-	runtimeMock2 := NewMockRuntimeInstance(ctrl)
-	runtimeMock3 := NewMockRuntimeInstance(ctrl)
+	errTest := errors.New("test error")
 
-	invalidTransaction := runtime.NewInvalidTransaction()
-	err := invalidTransaction.Set(runtime.Future{})
-	require.NoError(t, err)
-
-	type args struct {
-		peerID peer.ID
-		msg    *network.TransactionMessage
+	someHeader := &types.Header{
+		Number:    2,
+		StateRoot: common.Hash{2},
 	}
-	tests := []struct {
-		name             string
-		service          *Service
-		args             args
-		exp              bool
-		expErr           error
-		expErrMsg        string
-		ctrl             *gomock.Controller
-		mockNetwork      *mockNetwork
-		mockBlockState   *mockBlockState
-		mockStorageState *mockStorageState
-		mockTxnState     *mockTxnState
-		mockRuntime      *mockRuntime
+	someHeaderHash := someHeader.Hash()
+
+	testCases := map[string]struct {
+		serviceBuilder        func(ctrl *gomock.Controller) *Service
+		peerID                peer.ID
+		message               *network.TransactionMessage
+		propagateTransactions bool
+		errSentinel           error
+		errMessage            string
+		expectedMessage       *network.TransactionMessage
 	}{
-		{
-			name: "not synced",
-			mockNetwork: &mockNetwork{
-				IsSynced: false,
+		"not synced": {
```

Comment:
would be nice to change the tests names to `snake_case` pattern as they all appears as diffs

Reply:
Indeed, done 👍 

Created https://github.com/ChainSafe/gossamer/issues/3039 to do this for the entire codebase (with a regex and batch all)